### PR TITLE
Try adding ignoreScripts to fix pnpmfileChecksum update issue

### PR DESCRIPTION
**Pull request by sarayourfriend**
Also split config between explicit global/bot config and repository configuration

## Fixes

Part of #4586 by @sarayourfriend 

## Description

This PR does two things:

1. I noticed in the Renovate bot run that it complained about our combined repository and global configuration file. This wasn't ever intentional, I think just a misunderstanding of how to configure renovate for both contexts. I've split the file, so hopefully some of the debug output of the renovate bot will be a little easier to read (and mostly there will be a few less warnings in it). I also converted them to json5 so we can add explanatory comments to the configuration (Renovate supports json5 with comments).
2. The operative fix (hopefully) related to the linked issue is adding `ignoreScripts: false` to the monorepo's renovate configuration. [Check out the docs on this setting](https://docs.renovatebot.com/configuration-options/#ignorescripts): "Set this to false if allowScripts=true and you wish to run scripts when updating lock files.". We have allow scripts set to true, because `pnpmfile.cjs` needs to be part of the update process to lock TypeScript correctly (although, hopefully not for much longer if Nuxt 3 manages to liberate us to modern TS without this hack). In any case, we have that, and pnpmfile.cjs apparently is part of how pnpm computes the pnpmfileChecksum (check out this potentially related issue: https://github.com/pnpm/pnpm/issues/7951). Hopefully the issue is that renovate has thus far been ignore pnpmfile.cjs for lockfile updates? It seems like a stretch to me, like I'm somehow misunderstanding what Renovate means by "lockfile updates", but it's worth a shot to try and fix it, I think.

## Testing Instructions

Unfortunately it just needs to be merged and tried out the next time the bot runs. The validator should pass, of course.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (`./ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`./ov just catalog/generate-docs media-props`
      for the catalog or `./ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]
Really great, and if if fixes the checksum issue I'll consider that a bonus. LGTM.

File: .github/workflows/renovate.yml [hunk 1]
```diff
@@ -64,7 +64,11 @@ jobs:
 
       - uses: renovatebot/github-action@v40.1.12
         with:
-          configurationFile: .github/renovate.json
+          # Renovate recommends _not_ to name this one of the repository configuration file names:
+          # > It is recommended to not name it one of the repository configuration filenames listed in the Renovate Docs for Configuration Options.
+          # >
+          # > — https://github.com/renovatebot/github-action?tab=readme-ov-file#configurationfile
```
Comment (dhruvkb) [inline .github/workflows/renovate.yml:70]: I found this comment very hard to understand.

```suggestion
          # Renovate recommends _not_ to use any of
          # [these names](https://docs.renovatebot.com/configuration-options/).
```

**Review by dhruvkb — APPROVED** [review]
Hopefully the checksum issue is fixed. Nonetheless, the split config is more understandable now.

File: .github/workflows/renovate.yml (hunk 1 above)
Comment (sarayourfriend) [reply .github/workflows/renovate.yml]: Thanks for the suggestion!

**Comment by HonkingGoose** [comment]
Hi, you may want to subscribe to this discussion on the Renovate repository:

- https://github.com/renovatebot/renovate/discussions/30812

**Comment by sarayourfriend** [comment]
@obulat on that note, can we ditch the pnpmfile at this point, do we still need to manually bump the TS version after the Nuxt 3 migration?

**Pull request merged**